### PR TITLE
[Service Bus] calculate absolute expiry time based on time to live

### PR DESCRIPTION
**Pull request by swathipil**
fixes: #17970

### CODE REVIEW

File: sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py [hunk 1]
```diff
@@ -282,6 +284,13 @@ def time_to_live(self, value):
         else:
             self._amqp_header.time_to_live = int(value) * 1000
 
+        if self._amqp_header.time_to_live and self._amqp_header.time_to_live != MAX_DURATION_VALUE:
```
Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py:287]: does JS and .NET default `time_to_live` to MAX_DURATION_VALUE when constructing the message?

File: sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py [hunk 2]
```diff
@@ -282,6 +284,13 @@ def time_to_live(self, value):
...
+            self._amqp_properties.creation_time = int(utc_now().timestamp())
+            self._amqp_properties.absolute_expiry_time = min(
+                MAX_ABSOLUTE_EXPIRY_TIME,
```
Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py:290]: also do other languages default `absolute_expiry_time` to MAX_ABSOLUTE_EXPIRY_TIME?

File: sdk/servicebus/azure-servicebus/azure/servicebus/_common/constants.py [hunk 3]
```diff
@@ -5,6 +5,8 @@
 # -------------------------------------------------------------------------
 from enum import Enum
 
+from datetime import datetime
+from msrest.serialization import UTC
```
Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_common/constants.py:9]: are those two imports used?

File: sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py (hunk 1 above)
Comment (swathipil) [reply sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py:287]: yep! --> https://github.com/Azure/azure-sdk-for-js/blob/f53488f738ceb7c76fd1b76f67f36e1c8c868fc0/sdk/servicebus/service-bus/src/serviceBusMessage.ts#L323

File: sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py (hunk 2 above)
Comment (swathipil) [reply sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py:290]: yes --> https://github.com/Azure/azure-sdk-for-js/blob/f53488f738ceb7c76fd1b76f67f36e1c8c868fc0/sdk/servicebus/service-bus/src/serviceBusMessage.ts#L326

File: sdk/servicebus/azure-servicebus/azure/servicebus/_common/constants.py (hunk 3 above)
Comment (swathipil) [reply sdk/servicebus/azure-servicebus/azure/servicebus/_common/constants.py:9]: no, good catch!

File: sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py (hunk 2 above)
Comment (yunhaoling) [reply sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py:290]: I see, so this logic only applies when `time_to_live` is set

File: sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py (hunk 1 above)
Comment (yunhaoling) [reply sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py:287]: okay, I was wondering if JS/.NET would internally update amqp header time_to_live to be the MAX_DURATION_VALUE for outgoing messages, but they seems not

**Review by yunhaoling — APPROVED** [review]

**Pull request merged**